include/deal.II/matrix_free/evaluation_kernels.h

```diff
@@ -2928,9 +2933,42 @@ namespace internal
       Number *temp         = eval.get_scratch_data().begin();
       Number *scratch_data = temp + 3 * n_components * dofs_per_face;
 
-      FEFaceNormalEvaluationImpl<dim, fe_degree, Number>::template interpolate<
-        true,
-        false>(n_components, evaluation_flag, data, values_dofs, temp, face_no);
+      if (eval.get_dof_access_index() ==
+            MatrixFreeFunctions::DoFInfo::dof_access_cell &&
+          eval.get_is_interior_face() == false)
+        {
+          for (unsigned int v = 0; v < Number::size(); ++v)
+            {
+              // the loop breaks once an invalid_unsigned_int is hit for
+              // all cases except the exterior faces in the ECL loop (where
+              // some faces might be at the boundaries but others not)
+              if (eval.get_cell_ids()[v] == numbers::invalid_unsigned_int)
+                continue;
+
+              FEFaceNormalEvaluationImpl<dim, fe_degree, Number>::
+                template interpolate<true, false>(
+                  n_components,
+                  evaluation_flag,
+                  data,
+                  values_dofs,
+                  temp,
+                  eval.get_all_face_numbers()[v]);
+
+              for (unsigned int i = 0; i < 3 * n_components * dofs_per_face;
+                   ++i)
+                temp[i][v] = temp[i][v]; // TODO: we need a second buffer?
```

Comment:
Does `scratch_data` have enough space to use as buffer her?

Reply:
`3 * n_components * dofs_per_face` is exactly what we assume as memory before we let `scratch_data` begin. Since that additional array is long enough, we should have plenty of more space. But I guess this assignment is still to be updated as it is currently a self-assignment.